django_saml2_auth/user.py

```diff
@@ -227,16 +268,49 @@ def decode_jwt_token(jwt_token: str) -> Optional[str]:
         jwt_token (str): The token to decode
 
     Raises:
+        SAMLAuthError: Cannot decode JWT token. Specify an algorithm.
+        SAMLAuthError: Cannot decode JWT token. Specify a secret.
+        SAMLAuthError: Cannot decode JWT token. Specify a public key.
         SAMLAuthError: Cannot decode JWT token.
 
     Returns:
         Optional[str]: A user_id as str or None.
     """
-    jwt_secret = settings.SAML2_AUTH.get("JWT_SECRET")
     jwt_algorithm = settings.SAML2_AUTH.get("JWT_ALGORITHM")
+    jwt_secret = settings.SAML2_AUTH.get("JWT_SECRET")
+    jwt_public_key = settings.SAML2_AUTH.get("JWT_PUBLIC_KEY")
+
+    if not jwt_algorithm:
+        raise SAMLAuthError("Cannot create JWT token. Specify an algorithm.", extra={
```

Comment:
@mostafa it seems that these 3 errors should specify `decode` instead of `create` ?

Reply:
I also fixed other instances and made the error message more generic.